src/components/Profile/Activity.js

```diff
@@ -1,11 +1,35 @@
-import React from 'react'
-import styled from 'styled-components'
+import React, { useMemo } from 'react'
 import { Box, GU, Link, textStyle, useTheme } from '@1hive/1hive-ui'
 
+import ProposalIcon from '../ProposalIcon'
+import { useAppState } from '../../providers/AppState'
+import { useSupporterSubscription } from '../../hooks/useSubscriptions'
+
+import { convertToString } from '../../types'
+import { dateFormat } from '../../utils/date-utils'
+
 function Activity({ account }) {
   const theme = useTheme()
 
-  const myStakeHistory = []
+  const { honeypot } = useAppState()
+  const supporter = useSupporterSubscription(honeypot, account)
+
+  const dedupedStakes = useMemo(() => {
```

Comment:
what do you mean by deduped stakes here?

Reply:
Ah so for the activity, we are removing duplicated stakes to the same proposal.

This is to prevent for example if i staked, withdrew, and staked again on a proposal, it doesn't show in your activity that you did all those 3 things and just show that you staked on the proposal in general since there's no real value in showing all those 3.